xc/xc7/yosys/synth.tcl

```diff
@@ -21,10 +25,20 @@ if { $::env(USE_ROI) == "TRUE" } {
     read_verilog -lib $::env(TECHMAP_PATH)/iobs.v
 
     hierarchy -check -auto-top
+    update_pll_params
 
     # Start flow after library reading
-    synth_xilinx -vpr -flatten -abc9 -nosrl -noclkbuf -nodsp -iopad -nowidelut -run prepare:check
+    synth_xilinx -vpr -flatten -abc9 -nosrl -nodsp -iopad -nowidelut -run prepare:check
+}
+
+# Read the SDC file
+if { [info exists ::env(INPUT_SDC_FILE)] && $::env(INPUT_SDC_FILE) != "" } {
+  read_sdc $::env(INPUT_SDC_FILE)
+  propagate_clocks
```

Comment:
Why do you call it 3 times?

Reply:
That is a workaround to the fact that the buffer propagation doesn't reach a net that drives a PLL thus fails to find a PLL and perform the clock divider propagate. Basically, the first buffer propagation traverses through the IBUF, hence doesn't find the PLL. Only the second `propagate_clocks` triggers another buffer propagation from IBUF to BUFG and the following ClockDivider propagation does find the PLL and succeeds the clock divider propagation.
Long story short, that needs a fix, but was good enough to test.

Reply:
Ya, this needs to be handled internally.  3 iterations might be enough, 1 might be enough.